### PR TITLE
fund ropsten eth

### PR DESCRIPTION
**Pull request by LiorAgnin**
## Proposed changes

subj

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Refactor or a modification (non-breaking change)
- [x] New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: dapp/src/components/wizard/pages/ChooseNetwork.jsx [hunk 1]
```diff
@@ -1,17 +1,36 @@
...
+  return value === 'ropsten' && value === network && !hasEth && (
+    <button
+      onClick={(e) => {
+        e.preventDefault()
+        dispatch(fundEth(accountAddress))
+      }}
+      className='fund'
+      style={{ display: value === network && !hasEth ? 'block' : 'none' }}>
+      <span>
+        Click here to get 0.05 {value} eth
```
Comment (leonprou) [inline dapp/src/components/wizard/pages/ChooseNetwork.jsx:28]: Can you change to "Click here to get 0.05 {value} Ropsten ETH"

File: server/src/tasks/eth_funder.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+const config = require('config')
+const mongoose = require('mongoose')
+const { lockAccount, withAccount } = require('@utils/account')
+const { createNetwork } = require('@utils/web3')
+const { toWei, toChecksumAddress } = require('web3-utils')
+const EthFunding = mongoose.model('EthFunding')
+
+const ethFunder = withAccount(async (account, { accountAddress }) => {
+  try {
```
Comment (leonprou) [inline server/src/tasks/eth_funder.js:9]: no need to wrap in try/catch here

File: server/src/tasks/eth_funder.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    new EthFunding({
+      accountAddress: toChecksumAddress(accountAddress),
+      fundingDate: new Date()
+    }).save()
+    const bonus = config.get('bonus.eth.ropsten').toString()
+    const network = createNetwork('foreign', account)
+    const { web3 } = network
```
Comment (leonprou) [inline server/src/tasks/eth_funder.js:16]: worth checking here that the network is ropsten

File: server/src/tasks/eth_funder.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    await web3.eth.sendTransaction({
+      to: accountAddress,
+      from: account.address,
+      value: toWei(bonus),
+      nonce: account.nonce
+    })
```
Comment (leonprou) [inline server/src/tasks/eth_funder.js:22]: need to update `EthFunding.funded` to true

**Review by leonprou — CHANGES_REQUESTED** [review]
Looks good 👍 
see my comments.
Loved that you added the `EthFunding` model, we don't need to control the funding now but it's good to have the data.

File: dapp/src/components/wizard/pages/ChooseNetwork.jsx (hunk 1 above)
Comment (LiorAgnin) [reply dapp/src/components/wizard/pages/ChooseNetwork.jsx:28]: The value in ropsten :)

File: server/src/tasks/eth_funder.js (hunk 3 above)
Comment (LiorAgnin) [reply server/src/tasks/eth_funder.js:16]: Yes agreed, I'll change it.

File: server/src/tasks/eth_funder.js [hunk 5]
```diff
@@ -6,23 +6,22 @@ const { toWei, toChecksumAddress } = require('web3-utils')
...
-    throw e
-  }
+  const ethFunding = new EthFunding({
+    accountAddress: toChecksumAddress(accountAddress),
+    fundingDate: new Date()
+  }).save()
+  const bonus = config.get('bonus.eth.ropsten').toString()
+  const network = createNetwork('foreign', account)
+  const { web3, networkType } = network
+  if (networkType !== 'ropsten') throw Error('Fund ETH available only for ropsten')
```
Comment (leonprou) [inline server/src/tasks/eth_funder.js:16]: can you add brackets, like:
if () {

}

File: server/src/tasks/eth_funder.js [hunk 6]
```diff
@@ -6,23 +6,22 @@ const { toWei, toChecksumAddress } = require('web3-utils')
...
-    await web3.eth.sendTransaction({
-      to: accountAddress,
-      from: account.address,
-      value: toWei(bonus),
-      nonce: account.nonce
-    })
-  } catch (e) {
-    throw e
-  }
+  const ethFunding = new EthFunding({
```
Comment (leonprou) [inline server/src/tasks/eth_funder.js:9]: you need to add an `await` here cause there's a save at the end

**Review by leonprou — CHANGES_REQUESTED** [review]
Some small stuff I've found in the `ethFunder` func

**Comment by leonprou** [comment]
Also need to add again the ETH checks for Ropsten service

**Review by leonprou — APPROVED** [review]
👍🏻

**Pull request merged**